Wikipedia/Code/TalkPageContainerViewController.swift

```diff
@@ -191,6 +192,121 @@ private extension TalkPageContainerViewController {
+            self.openURLInSafari(url: absoluteURL)
+        }
+        let talkAction = UIAlertAction(title: WMFLocalizedString("talk-page-link-user-action-sheet-app", value: "View User Talk page in app", comment: "Title of action sheet button that takes user to a user talk page in the app after tapping a user page link in talk pages."), style: .default) { (_) in
+            
+            let title = absoluteURL.lastPathComponent
+            if let firstColon = title.range(of: ":") {
+                var titleWithoutNamespace = title
+                titleWithoutNamespace.removeSubrange(title.startIndex..<firstColon.upperBound)
+                let titleWithTalkPageNamespace = TalkPageType.user.titleWithCanonicalNamespacePrefix(title: titleWithoutNamespace, siteURL: siteURL)
+                self.pushTalkPage(title: titleWithTalkPageNamespace, siteURL: siteURL)
+            }
+        }
+        let cancelAction = UIAlertAction(title: CommonStrings.cancelActionTitle, style: .cancel, handler: nil)
+        
+        alertController.addAction(safariAction)
+        alertController.addAction(talkAction)
+        alertController.addAction(cancelAction)
+        
+        present(alertController, animated: true, completion: nil)
+    }
+    
+    func openURLInSafari(url: URL) {
+        UIApplication.shared.open(url, options: [:], completionHandler: nil)
+    }
+    
+    func toggleLinkDeterminationState(loadingViewController: FakeLoading & ViewController, shouldDisable: Bool) {
+        
+        if shouldDisable {
+            loadingViewController.fakeProgressController.start()
+        } else {
+            loadingViewController.fakeProgressController.stop()
+        }
+        
+        loadingViewController.scrollView?.isUserInteractionEnabled = !shouldDisable
+        loadingViewController.navigationItem.rightBarButtonItem?.isEnabled = !shouldDisable
+    }
+    
+    func tappedLink(_ url: URL, loadingViewController: FakeLoading & ViewController) {
+        guard let absoluteURL = absoluteURL(for: url) else {
+            showNoInternetConnectionAlertOrOtherWarning(from: TalkPageError.unableToDetermineAbsoluteURL)
+            return
+        }
+        
+        toggleLinkDeterminationState(loadingViewController: loadingViewController, shouldDisable: true)
+        
+        self.dataStore.articleSummaryController.updateOrCreateArticleSummariesForArticles(withURLs: [absoluteURL]) { [weak self, weak loadingViewController] (articles, error) in
+            
+            guard let self = self,
+            let loadingViewController = loadingViewController else {
+                return
+            }
+            
+            self.toggleLinkDeterminationState(loadingViewController: loadingViewController, shouldDisable: false)
+            
+            if error != nil {
+                self.openURLInSafari(url: absoluteURL)
```

Comment:
If this is a network error, might be better to show it instead of bumping to Safari

Reply:
@joewalsh true - I had noticed global user pages were returning 404s from the summary endpoint, which had me thinking maybe it would be good to give Safari a second chance to resolve it in case there are other cases like that. The 404 case gets caught in line [306](https://github.com/wikimedia/wikipedia-ios/blob/2b1aacaacd900c7db257fe656aa6a6722985155d/Wikipedia/Code/TalkPageContainerViewController.swift#L306) though, so I can easily change this without breaking that.